test_scripts/UnknownEnums/006_unkown_enum_in_array_of_structs.lua

```diff
@@ -0,0 +1,167 @@
+  mediaTrack = "mediaTrack_text",
+  alignment = "CENTERED",
+  graphic = {
+    imageType = "DYNAMIC",
+    value = "icon.png"
+  },
+  secondaryGraphic = {
+    imageType = "DYNAMIC",
+    value = "icon.png"
+  },
+  metadataTags = {
+    mainField1 = { "mediaTitle" },
+    mainField2 = { "mediaArtist" },
+    mainField3 = { "mediaAlbum" },
+    mainField4 = { "mediaYear" },
+  },
+  softButtons = { 
+    {
+    softButtonID = 1,
+    type = "UNKNOWN"
+    },
+    {
+    softButtonID = 2,
+    type = "TEXT",
+    text = "Hello"
+    },
+  }
+}
+
+local responseUiParams = {
+  showStrings = {
+    {
+      fieldName = "mainField1",
+      fieldText = requestParams.mainField1,
+      fieldTypes = requestParams.metadataTags.mainField1
+    },
+    {
+      fieldName = "mainField2",
+      fieldText = requestParams.mainField2,
+      fieldTypes = requestParams.metadataTags.mainField2
+    },
+    {
+      fieldName = "mainField3",
+      fieldText = requestParams.mainField3,
+      fieldTypes = requestParams.metadataTags.mainField3
+    },
+    {
+      fieldName = "mainField4",
+      fieldText = requestParams.mainField4,
+      fieldTypes = requestParams.metadataTags.mainField4
+    },
+    {
+      fieldName = "templateTitle",
+      fieldText = requestParams.templateTitle
+    },
+    {
+      fieldName = "mediaClock",
+      fieldText = requestParams.mediaClock
+    },
+    {
+      fieldName = "mediaTrack",
+      fieldText = requestParams.mediaTrack
+    },
+    {
+      fieldName = "statusBar",
+      fieldText = requestParams.statusBar
+    }
+  },
+  alignment = requestParams.alignment,
+  graphic = {
+    imageType = requestParams.graphic.imageType,
+    value = common.getPathToFileInAppStorage(requestParams.graphic.value)
+  },
+  secondaryGraphic = {
+    imageType = requestParams.secondaryGraphic.imageType,
+    value = common.getPathToFileInAppStorage(requestParams.secondaryGraphic.value)
+  },
+  softButtons = {{
+    softButtonID = 2,
+    type = "TEXT",
+    text = "Hello"
+  }}
+}
+
+local allParams = {
+  requestParams = requestParams,
+  responseUiParams = responseUiParams,
+}
+
+--[[ Local Functions ]]
+local function show(pParams)
+  local cid = common.getMobileSession():SendRPC("Show", pParams.requestParams)
+  pParams.responseUiParams.appID = common.getHMIAppId()
+  common.getHMIConnection():ExpectRequest("UI.Show", pParams.responseUiParams)
+  :Do(function(_, data)
+      common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", { })
+    end)
+  :ValidIf(function(_, data)
+      return #data.params["softButtons"] == 1
+    end)
+  common.getMobileSession():ExpectResponse(cid, { 
+    success = true, 
+    resultCode = "WARNINGS"
+  }):ValidIf(function(_, data)
+    return string.match(data.payload.info, "softButtons")
+  end)
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Update Preloaded PT", common.updatePreloadedPT)
```

Comment:
It seems updating of preloaded file is not required for this case.

Reply:
Addressed in 07b462b